lib/freeclimb/utils/request_verifier.rb

```diff
@@ -0,0 +1,58 @@
+module Freeclimb
+    class RequestVerifier
+        @@DEFAULT_TOLERANCE = 5 * 60 * 1000
+        attr_accessor :info
+
+        def verifyRequestSignature(requestBody, requestHeader, signingSecret, tolerance=DEFAULT_TOLERANCE)
+            checkRequestBody(requestBody)
+            checkRequestHeader(requestHeader)
+            checkSigningSecret(signingSecret)
+            checkTolerance(tolerance)
+            @info = Freeclimb::SignatureInformation.new(requestHeader)
+            verifyTolerance(@info, tolerance)
+            verifySignature(@info, requestBody, signingSecret)
+        end
+        
+        def checkRequestBody(requestBody)
+            if requestBody == "" or requestBody == nil
+                raise 'Request Body cannot be empty or null'
+            end
+        end
+
+        def checkRequestHeader(requestHeader)
+            if requestHeader == "" or requestHeader == nil
+                raise 'Error with request header, Request header is empty'
+            elsif ! requestHeader.include? "t"
+                raise 'Error with request header, timestamp is not present'
+            elsif ! requestHeader.include? "v1"
+                raise 'Error with request header, signatures are not present'
+            end
+        end
+
+        def checkSigningSecret(signingSecret)
+            if signingSecret == "" or signingSecret == nil
+                raise 'Signing secret cannot be empty or null'
+            end
+        end
+
+        def checkTolerance(tolerance)
+            fixnum = (2**(0.size * 8 -2) -1)
```

Comment:
This is not an accurate way to check if the number is exceeding the max integer. Running this locally and I can increase the value of it to a much larger value than whatever the result of this is.

Reply:
So from what I've been researching, it seems that integers don't have a maximum value in Ruby: https://stackoverflow.com/a/47894026

So what I'm thinking is creating a class for the maximum integer so then we can use it within the two util files

I can use this as reference and create an class for this:
https://gist.github.com/pithyless/9738125

Reply:
If there is no limit, then there is no reason to validate against it. Simply checking that it is a positive integer is enough

Reply:
will change the conditional in latest commit